scripts/cromwell/analyze_resource_acquisition.py

```diff
@@ -0,0 +1,460 @@
+	
+	call_metadata['vm'] = call_metadata.vm_delta.cumsum()
+	call_metadata['cpu_all'] = call_metadata.cpu_all_delta.cumsum()
+	call_metadata['cpu_preemptible'] = call_metadata.cpu_preemptible_delta.cumsum()
+	call_metadata['cpu_nonpreemptible'] = call_metadata.cpu_nonpreemptible_delta.cumsum()
+	call_metadata['memory'] = call_metadata.memory_delta.cumsum()
+	call_metadata['ssd'] = call_metadata.ssd_delta.cumsum()
+	call_metadata['hdd'] = call_metadata.hdd_delta.cumsum()
+
+	return call_metadata
+
+def plot_resources_time(df, title_name, output_name):
+	"""
+	Modified from: https://github.com/broadinstitute/dsde-pipelines/blob/master/scripts/quota_usage.py
+	"""
+	print("Writing " + output_name, file = sys.stderr)
+	colors = {
+		"vm": "#006FA6", #blue 
+		"cpu_all": "black",
+		"cpu_preemptible": "#10a197", #turquoise 
+		"cpu_nonpreemptible": "#A30059", #dark pink
+		"memory": "#FF4A46", #coral red
+		"hdd": "#72418F", #purple
+		"ssd": "#008941", #green
+	}
+	LABEL_SIZE = 17
+	TITLE_SIZE = 20
+	TICK_SIZE = 15
+
+	fig,ax = plt.subplots(4,1, figsize = (14,26), sharex = True)
+	ax[0].set_title(title_name + "Resource Acquisition Over Time", fontsize=TITLE_SIZE)
+
+	ax[0].plot(df['seconds'], df['vm'], color = colors["vm"])
+	ax[0].set_ylabel("VMs", fontsize = LABEL_SIZE)
+	plt.setp(ax[0].get_yticklabels(), fontsize = TICK_SIZE)
+
+	ax[1].plot(df['seconds'], df['cpu_all'], color = colors["cpu_all"], linewidth = 2, label = "All")
+	ax[1].plot(df['seconds'], df['cpu_preemptible'], color = colors["cpu_preemptible"], linestyle = "dashed", label = "Preemptible")
+	ax[1].plot(df['seconds'], df['cpu_nonpreemptible'], color = colors["cpu_nonpreemptible"], linestyle = "dashed", label = "Non-preemptible")
+	ax[1].set_ylabel("CPU Cores", fontsize = LABEL_SIZE)
+	plt.setp(ax[1].get_yticklabels(), fontsize = TICK_SIZE)
+	ax[1].legend(loc = "upper right", title = "CPU Types", fontsize = TICK_SIZE, title_fontsize = TICK_SIZE)
+
+	ax[2].plot(df['seconds'], df['memory'], color = colors["memory"])
+	ax[2].set_ylabel("RAM (GiB)", fontsize = LABEL_SIZE)
+	plt.setp(ax[2].get_yticklabels(), fontsize = TICK_SIZE)
+
+	ax[3].plot(df['seconds'], df['hdd'], color = colors["hdd"], label = "HDD")
+	ax[3].plot(df['seconds'], df['ssd'], color = colors["ssd"], label = "SSD")
+	ax[3].set_ylabel("Disk Memory (GiB)", fontsize = LABEL_SIZE)
+	plt.setp(ax[3].get_yticklabels(), fontsize = TICK_SIZE)
+	ax[3].legend(loc = "upper right", title = "Disk Types", fontsize = TICK_SIZE, title_fontsize = TICK_SIZE)
+
+	formatter = matplotlib.ticker.FuncFormatter(lambda x,pos: str(datetime.timedelta(seconds = x)))
+	ax[3].xaxis.set_major_formatter(formatter) 
+	plt.setp(ax[3].get_xticklabels(), rotation = 15, fontsize = TICK_SIZE)
+	ax[3].set_xlabel("Time", fontsize = LABEL_SIZE)
+	
+	fig.savefig(output_name, bbox_inches='tight')
+
+def write_resources_time_table(call_metadata, table_file):
+	print("Writing " + table_file, file = sys.stderr)
+	call_metadata.to_csv(
```

Comment:
The time format looks like this for the example I ran: `0 days 00:00:00`. Can you set the `date_format` parameter for `to_csv` to use the ISO 8601 format? This is what Cromwell uses and doesn't contain any whitespace.

Reply:
Done. Although I don't anticipate this table being particularly useful, which is why it isn't printed by default. I was considering just removing the option to write it to a file - it's used to create the plot, but it's not very human-readable.